merino-web/src/endpoints/suggest.rs

```diff
@@ -169,3 +174,50 @@ impl<'a> Serialize for SuggestionWrapper<'a> {
         generated.serialize(serializer)
     }
 }
+
+/// Log a suggestion request, respecting the log_query setting passed, and
+/// formatting all fields in a way that is helpful to our downstream log
+/// handlers. This primarily means that fields don't generate internal
+/// stringified JSON.
+fn safe_log_request(
+    log_query: bool,
+    request: &SuggestionRequest,
+    query_params: &SuggestQueryParameters,
+) {
+    let none = "none".to_string();
+    let country = request.country.as_ref().unwrap_or(&none);
+    let region = request.region.as_ref().unwrap_or(&none);
+    let city = request.city.as_ref().unwrap_or(&none);
+    let dma = request.dma.map_or_else(|| none.clone(), |v| v.to_string());
+    let query = if log_query {
+        request.query.as_str()
+    } else {
+        ""
+    };
+    let requested_providers = query_params.providers.as_ref().map_or_else(
+        || "".to_string(),
+        |ps| {
+            ps.iter()
+                .map(|p| p.as_str())
+                .collect::<Vec<_>>()
+                .as_slice()
+                .join(",")
+        },
+    );
+
+    tracing::info!(
+        accepts_english = ?request.accepts_english,
+        %city,
+        %country,
+        os_family = %request.device_info.os_family,
+        form_factory = %request.device_info.form_factor,
+        browser = %request.device_info.browser,
+        %dma,
+        %region,
+        %query,
+        client_variants = %query_params.client_variants.join(","),
+        %requested_providers,
+        r#type = "web.suggest.request",
```

Comment:
cc @jklukas: this is the type to route log lines on. Log lines with this types _may_ contain search queries.